keps/sig-api-machinery/4020-unknown-version-interoperability-proxy/README.md

```diff
@@ -377,7 +378,24 @@ upon bootstrap
 
 ### Discovery Merging
 
-TODO: detailed description of discovery merging. (not scheduled until beta.)
+A new handler will be introduced to serve a consolidated discovery document. This handler is the core of the merged discovery feature, which combines local API server discovery with information from peer API servers in the cluster. This is achieved by extending the functionality of the existing aggregated discovery endpoints, and no new endpoints will be introduced.
+
+The new handler is responsible for the following actions:
+
+* Document Generation: The handler will intelligently merge local discovery data with information from the PeerAggregatedDiscovery cache to create a single, comprehensive view of all available API groups and resources
+* Client Negotiation: The handler will interpret a new `profile` parameter in the `Accept` header. By default, it will serve a merged discovery document. A client can explicitly request an unmerged, local-only discovery response by setting `profile=unmerged`. This capability is essential because it allows the API server to fetch the unmerged discovery documents from its peers. The information from these requests is then used to populate the API server's PeerAggregatedDiscovery cache. This cache is fundamental for two main reasons: it is the critical data source for building the consolidated, merged discovery response, and it is used to determine which peer API server is capable of handling a proxied resource request, making the proxying logic possible.
+* Backward Compatibility: The handler will ensure that un-aggregated discovery requests continue to function as before. The entire feature is controlled by a feature gate, allowing for a safe, gradual rollout. To maintain interoperability in a mixed-version cluster, a key design point is the use of the Accept header to distinguish between different request types.
+  * Peer-to-Peer Discovery: When a newer API server, with the feature enabled, makes a discovery call to an older peer to populate its local cache, it uses a content type variant `(application/json;g=apidiscovery.k8s.io;v=v2;as=APIGroupDiscoveryList;...;profile=unmerged)`. This allows the older peer to fall back to its standard discovery behavior, ensuring the request succeeds and the newer API server can collect the necessary unmerged data.
```

Comment:
Let's also clarify that it will ask for `application/json;...;profile=unmerged, application/json;...;q=0.9` this way if the request is handled by an apiserver pre-dating this feature, it will correctly serve discovery normally (without merging) even though it's not aware of the new profile.